src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/DictionaryModelBinder.cs

```diff
@@ -113,10 +113,7 @@ public class DictionaryModelBinder<TKey, TValue> : CollectionModelBinder<KeyValu
                 return collection.ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
             }
 
-            var newCollection = CreateInstance(targetType);
-            CopyToModel(newCollection, collection);
-
-            return newCollection;
+            return base.ConvertToCollectionType(targetType, collection);
```

Comment:
Just avoiding some code duplication. Has the slight side effect of allowing this binder to handle `IList<KeyValuePair<K,V>>` but that's not a big deal either way.